website/docs/docs/cloud/manage-access/set-up-bigquery-oauth.md

```diff
@@ -89,7 +89,54 @@ Select **Allow**. This redirects you back to <Constant name="cloud" />. You are
+6. Click **Add** next to Application ID URI. The field will automatically populate. 
+7. Click **Save**.
+
+Workload Identity Federation utilizes a machine-to-machine OAuth flow that is unattended by the user; as such, a redirect URI won't need to be set for the application. Step three in this section is crucial because it determines the audience for tokens issued from the app and informs the workpool in GCP whether the calling application has permission to access the resources guarded by the workpool.
+
+- **Related documentation:** [GCP &mdash; Prepare your external identity provider](https://cloud.google.com/iam/docs/workload-identity-federation-with-other-clouds#create)
+
+### 2. Create a Workpool and Workpool Provider in GCP
+
+1. In your GCP account's main menu, navigate to **IAM & Admin** and click the **Workload Identity Federation** option (not to be confused with the **Workforce Identity Federation** option directly adjacent). 
+2. If you haven’t created a workpool yet, click **Get started** or create a new workpool (button near the top of the page).
+3. Give the workpool a name and description. Per the [GCP documentation](https://cloud.google.com/iam/docs/workload-identity-federation#pools), a new pool should be created for each non-Google Cloud environment that needs to access Google Cloud resources, such as development, staging, or production environments. The workpool should be named accordingly to make it easily identifiable in the future.
+4. When creating your provider:
+    - Set the type of the provider to **OpenID Connect (OIDC)**.
+    - Name the provider something identifiable, like `Entra ID`.
+    - Set the URL to https://sts.windows.net/YOUR_TENANT_ID/. This can be found in the token itself, if you decode it via jwt.io. You can also see a reference to the expected issuer URL for Entra in the [GCP documentation for WIF](https://cloud.google.com/iam/docs/workload-identity-federation-with-other-clouds#create_the_workload_identity_pool_and_provider).
+        - The tenant (provider) ID can be found in the app registration created in [section 1 of these instructions](#1-set-up-entra-id); it's called **Directory (tenant) ID** and can be found in the overview section for the application.
+    - For **Audiences**, select **Allowed Audiences** and set the value to the **Application ID URI** that was defined for your Entra ID app.
+5. Click **Continue**.
+6. Under **Configure provider attributes**, set the mapping for `google.subject` to `assertion.sub`.
+7. Click **Save**.
+
+### 3. Service Account Impersonation
+
+A workpool either uses a service account or is granted direct resource access to determine which resources a caller has access to. The [GCP documentation](https://cloud.google.com/iam/docs/workload-identity-federation-with-other-clouds#access) provides more detailed information on configuring both for your workpool. We chose the service account approach in our implementation because it offered greater flexibility.
```

Comment:
```suggestion
A workpool either uses a service account or is granted direct resource access to determine which resources a caller can access. The [GCP documentation](https://cloud.google.com/iam/docs/workload-identity-federation-with-other-clouds#access) provides more detailed information on configuring both for your workpool. We chose the service account approach in our implementation because it offered greater flexibility.
```